moveit_core/online_signal_smoothing/test/test_acceleration_filter.cpp

```diff
@@ -0,0 +1,196 @@
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include <gtest/gtest.h>
+#include <moveit/online_signal_smoothing/acceleration_filter.h>
+#include <moveit/utils/robot_model_test_utils.h>
+
+class AccelerationFilterTest : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+  };
+
+  void set_limit(std::optional<Eigen::VectorXd> acceleration_limit)
+  {
+    const std::vector<moveit::core::JointModel*> joint_models = robot_model_->getJointModels();
+    auto joint_model_group = robot_model_->getJointModelGroup(move_group_name_);
+    size_t ind = 0;
+    for (auto& joint_model : joint_models)
+    {
+      if (!joint_model_group->hasJointModel(joint_model->getName()))
+      {
+        continue;
+      }
+      std::vector<moveit_msgs::msg::JointLimits> joint_bounds_msg(joint_model->getVariableBoundsMsg());
+      for (auto& joint_bound : joint_bounds_msg)
+      {
+        joint_bound.has_acceleration_limits = acceleration_limit.has_value();
+        if (joint_bound.has_acceleration_limits)
+        {
+          joint_bound.max_acceleration = acceleration_limit.value()[ind];
+        }
+      }
+      joint_model->setVariableBounds(joint_bounds_msg);
+      ind++;
+    }
+  }
+
+protected:
+  moveit::core::RobotModelPtr robot_model_;
+  std::string move_group_name_ = "panda_arm";
+};
+
+TEST_F(AccelerationFilterTest, FilterInitialize)
+{
+  online_signal_smoothing::AccelerationLimitedPlugin filter;
+
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("AccelerationFilterTest");
+  node->declare_parameter<std::string>("move_group_name", move_group_name_);
+
+  // fail because the number of joint is wrong
+  EXPECT_FALSE(filter.initialize(node, robot_model_, 3));
+
+  // fail because the update_rate parameter is not set
+  EXPECT_FALSE(filter.initialize(node, robot_model_, 7));
+  node->declare_parameter<double>("update_rate", 0.01);
+
+  // fail because the robot does not have acceleration limits
+  set_limit({});
+  EXPECT_FALSE(filter.initialize(node, robot_model_, 7));
+
+  // succeed with acceleration limits
+  Eigen::VectorXd acceleration_limit = 1.2 * Eigen::VectorXd::Ones(7);
+  set_limit(acceleration_limit);
+  EXPECT_TRUE(filter.initialize(node, robot_model_, 7));
+}
+
+TEST_F(AccelerationFilterTest, FilterDoSmooth)
+{
+  online_signal_smoothing::AccelerationLimitedPlugin filter;
+
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("AccelerationFilterTest");
+  node->declare_parameter<std::string>("move_group_name", move_group_name_);
+  double update_rate = 1.0;
```

Comment:
this can be `const` -- same in the other cases

Reply:
Done